### PR TITLE
Mttl 2835 - Implement CloseProcess trigger 

### PR DESCRIPTION
**Pull request by notatiyyah**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-2835

## Describe this PR

- Add CloseProcess & publish ProcessClosed events when it is triggered
- Update some state transitions that were mistakenly triggering ProcessClosed events instead of ProcessUpdated events (e.g. on manual checks failing)
- Fix bug in the E2E tests where SQS messages were not correctly picked up by the SnsFixture, causing falsely passing tests

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

The bug with the events in the E2E tests likely have affected other APIs, which will need to be updated.

### CODE REVIEW

**Review by rcmsoftwaredevelopment — APPROVED** [review]

File: ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs [hunk 1]
```diff
@@ -4,10 +4,11 @@ namespace ProcessesApi.V1.Domain
 {
     public static class SoleToJointStates
     {
+        public const string ProcessClosed = "ProcessClosed";
```
Comment (humulla) [inline ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs:7]: Can this state be a part of SharedProcessConstants?

File: ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs [hunk 2]
```diff
@@ -4,10 +4,11 @@ namespace ProcessesApi.V1.Domain
...
+        public const string ProcessCancelled = "ProcessCancelled";
```
Comment (humulla) [inline ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs:8]: Can this state be a part of SharedProcessConstants?

File: ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs [hunk 3]
```diff
@@ -20,8 +21,9 @@ public static class SoleToJointStates
 
     public static class SoleToJointPermittedTriggers
     {
-        public const string CheckAutomatedEligibility = "CheckAutomatedEligibility";
         public const string CancelProcess = "CancelProcess";
+        public const string CloseProcess = "CloseProcess";
```
Comment (humulla) [inline ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs:25]: Can this trigger be a part of SharedProcessConstants?

File: ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs [hunk 4]
```diff
@@ -20,8 +21,9 @@ public static class SoleToJointStates
...
+        public const string CheckAutomatedEligibility = "CheckAutomatedEligibility";
```
Comment (humulla) [inline ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs:26]: Can this trigger be a part of SharedProcessConstants?

**Review by humulla — COMMENTED** [review]
Looks good to be, there are a few refactoring comments.  Also please update the Hackney Core testing NuGet package version so that we are not merging the preview one.

File: ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs (hunk 3 above)
Comment (notatiyyah) [reply ProcessesApi/V1/Domain/SoleToJoint/SoleToJointConstants.cs:25]: Not really.. the issue is with the permittedTriggers list, if it's in the shared constants class it won't show up in the permittedTriggers array.

**Review by humulla — APPROVED** [review]

**Pull request merged**